pkg/ottl/ottlfuncs/func_replace_pattern_test.go

```diff
@@ -45,15 +45,57 @@ func Test_replacePattern(t *testing.T) {
 		{
 			name:    "replace regex match (with hash function)",
 			target:  target,
+			pattern: `passwd\=([^\s]*)(\s?)`,
+			replacement: ottl.StandardStringGetter[pcommon.Value]{
+				Getter: func(context.Context, pcommon.Value) (any, error) {
+					return "$1", nil
+				},
+			},
+			function: optionalArg,
+			want: func(expectedValue pcommon.Value) {
+				expectedValue.SetStr("application hash(sensitivedtata) otherarg=notsensitive key1 key2")
+			},
+		},
+		{
+			name:    "replace regex match (capture group without $1)",
+			target:  target,
+			pattern: `passwd\=([^\s]*)(\s?)`,
+			replacement: ottl.StandardStringGetter[pcommon.Value]{
+				Getter: func(context.Context, pcommon.Value) (any, error) {
+					return "test", nil
+				},
+			},
+			function: optionalArg,
+			want: func(expectedValue pcommon.Value) {
+				expectedValue.SetStr("application hash(test) otherarg=notsensitive key1 key2")
+			},
+		},
+		{
+			name:    "replace regex match (no capture group with $1 and hash function)",
+			target:  target,
 			pattern: `passwd\=[^\s]*(\s?)`,
```

Comment:
```suggestion
			pattern: `passwd\=[^\s]*\s?`,
```
The test name says there shouldn't be a capture group in the pattern.